spec/Vehicle/Connectivity.vspec

```diff
@@ -21,3 +21,130 @@ IsConnectivityAvailable:
   comment: This signal can be used by onboard vehicle services to decide what features that
            shall be offered to the driver, for example disable the 'check for update' button if vehicle does
            not have connectivity.
+           
+ConnectivityType:
+  datatype: string[]
+  type: attribute
+  allowed: [
+    'CELLULAR',
+    'WIFI'
+    ]
+  description: Type of connectivity currently available.
+  
+CurrentDownload:
+  datatype: float
+  type: sensor
+  unit: MiB/s
+  description: Download bandwidth currently in-use.
+
+CurrentUpload:
+  datatype: float
+  type: sensor
+  unit: MiB/s
+  description: Upload bandwidth currently in-use.
+
+Cellular:
+  type: branch
+  description: Properties related to cellular connectivity.
+  
+Cellular.SignalStrength:
+  type: sensor
+  datatype: int8
+  unit: dBm
+  description: Cellular connectivity signal strength.
+  
+Cellular.NetworkType:
+  datatype: string[]
+  type: sensor
+  allowed: [
+    '2G',
+    '3G',
+    '4G',
+    '5G'
+    ]
+  description: Type of cellular network that is currently in-use.
+  comment:      There are a lot of subtypes that fall under the general '~G' types, but these aren't expanded upon here.
+  
+Cellular.NetworkSubtype:
+  datatype: string
+  type: sensor
+  description: Cellular network technology that is currently in-use.
+  comment:     For example, LTE and WiMAX both fall under the '4G' umbrella but are substantially different.
+  
+Cellular.Carrier:
+  datatype: string
+  type: sensor
+  description: Carrier that is currently in-use.
+
+Cellular.DataUsage:
+  type: branch
+  description: Information on cellular data usage.
+  
+Cellular.DataUsage.Limit:
+  type: attribute
+  unit: MiB
+  datatype: uint32
+  min: 0
+  description: Data usage limit in megabytes.
+  
+Cellular.DataUsage.Current:
+  type: sensor
+  unit: MiB
+  datatype: uint32
+  min: 0
+  description: Amount of data used this cycle.
+  
+Cellular.DataUsage.CycleLength:
+  type: attribute
+  unit: day
+  datatype: uint16
+  min: 0
+  description: Length of a data cycle.
+  comment:     Defines the period where the current data usage is relevant.
+  
+Cellular.DataUsage.CycleReset:
+  type: attribute
+  unit: day
+  datatype: uint16
+  min: 0
+  description: Time until the next data cycle starts.
```

Comment:
I don't love this implementation, but I can't think of a simple alternative other than something like NextCycle, represented by a Unix timestamp.

Reply:
One could alternatively have "CycleStart" (as unix timestamp or ISO 8601 string). Together with CycleLength you then now when next reset should happen.

Reply:
I like that solution much better, and will implement it.